java/test/org/openqa/selenium/grid/data/DefaultSlotMatcherTest.java

```diff
@@ -29,6 +30,27 @@ class DefaultSlotMatcherTest {
 
   private final DefaultSlotMatcher slotMatcher = new DefaultSlotMatcher();
 
+  @Test
+  void testBrowserVersionMatch() {
+    assertThat(browserVersionComparator("", "")).isEqualTo(0);
+    assertThat(browserVersionComparator("", "130.0")).isEqualTo(-1);
+    assertThat(browserVersionComparator("131.0.6778.85", "131")).isEqualTo(0);
+    assertThat(browserVersionComparator("131.0.6778.85", "131.0")).isEqualTo(0);
+    assertThat(browserVersionComparator("131.0.6778.85", "131.0.6778")).isEqualTo(0);
+    assertThat(browserVersionComparator("131.0.6778.85", "131.0.6778.95")).isEqualTo(1);
+    assertThat(browserVersionComparator("130.0", "130.0")).isEqualTo(0);
+    assertThat(browserVersionComparator("130.0", "130")).isEqualTo(0);
+    assertThat(browserVersionComparator("130.0.1", "130")).isEqualTo(0);
+    assertThat(browserVersionComparator("130.0.1", "130.0.1")).isEqualTo(0);
+    assertThat(browserVersionComparator("133.0a1", "133")).isEqualTo(0);
```

Comment:
A common test for comperators is to flip the input value to see if the compare does return the reversed (or in this case same) value for both calls.
e.g. `assertThat(browserVersionComparator("133", "133.0a1")).isEqualTo(0);` does fail.

It might be best to only compare the minimum length of the arrays resulting from splitting at '.'